File: csharp/ChainOfResponsibility-Pets/Cat.cs
public class Cat : Pet
{
    public override string HandleTrap(Bait bait)
    {
        if (bait == Bait.SAUSAGE)
            return "purring cat";
        else
            return nextPet.HandleTrap(bait);
    }

    public Cat(Pet nextPet) : base(nextPet)
    {
    }
}

File: java/src/main/java/codingdojo/Cat.java
package codingdojo;

public class Cat extends Pet {
    public Cat(Pet nextPet) {
        super(nextPet);
    }

    @Override
    public String handleTrap(Bait bait) {
        if (bait == Bait.SAUSAGE)
            return "purring cat";
        else
            return nextPet.handleTrap(bait);
    }
}
